### DISCUSSION TITLE
Symfony 3.4 comptibility with PHP 8.2

### DISCUSSION THREAD

**Discussion by alidhibi** [symfony/symfony#52687] [Q&A]
Hi everyone,
I'm wondering if there is away for making Symfony 3.4 working on PHP 8.2.
We have a legacy code and we can note rewrite all the code and migrate to latest Symfony version.
All my new project are made within the last version but i do not have enough time to migrate my old application.
I'tried to use PHP Rector and php-cs to make changes but without hope ^_^' 
Can some one give me and advice?
Thanks in advance :)

**Comment by MatTheCat** [comment]
If you cannot update your apps nor downgrade their PHP version, you can still work around this by adding the `--ignore-platform-req=php` flag to `composer install`.

There is no guarantee Symfony 3.4's code will work on PHP 8.2 though, so I would not recommend it.

**Reply by alidhibi** [reply]
Yes, I was able to create a version of composer.json that installs without errors related to the PHP version. Now the biggest challenge is to make everything work and validate the unit tests.

**Comment by ghost** [comment]
IIRC support for new PHP versions is provided during the bug fixes period. SF 3.4 bug fixes ended on November 2020. PHP 8.0.0. was releases at the same time. You can search https://github.com/symfony/symfony/blob/184bd684f9dc243c37faa79b8d90f506b3fc4174/CHANGELOG-3.4.md for commits with `php 8` in the title (you can forget about PHP 8.1+)
I think however, Symfony components are the least of your problems, because this is a very well supported framework. Problems will start with other libraries that do not have such long support period for given major versions and have less maintainers/contributors.

**Reply by alidhibi** [reply]
Indeed, there are quite a few components that are not compatible with PHP 8.2. In this case, I went through an update for some and a code migration with Rector for those that are no longer maintained. Now I have some issues in the framework bundle, and I continue to try to resolve them. Thank you anyway for the feedback.

**Reply by tacman** [reply]
don't forget about composer recipes -- I've found that running those on each bundle helps with the migration process, especially for things like framework, which changed the console and index.php scripts.
